components/pkg-export-helm/src/chart.rs

```diff
@@ -0,0 +1,202 @@
+                return Err(Error::InvalidBindSpec(bind.to_string()).into());
+            }
+
+            let name_var = format!("bindName{}", i);
+            let service_var = format!("bindService{}", i);
+            let group_var = format!("bindGroup{}", i);
+            i = i + 1;
+
+            values.add_entry(&name_var, split[0]);
+            values.add_entry(&service_var, split[1]);
+            values.add_entry(&group_var, split[2]);
+
+            let json = json!({
+                "name": format!("{{{{.Values.{}}}}}", name_var),
+                "service": format!("{{{{.Values.{}}}}}", service_var),
+                "group": format!("{{{{.Values.{}}}}}", group_var),
+            });
+
+            binds.push(json);
+        }
+
+        let manifest_template = ManifestJson {
+            main: main,
+            binds: binds,
+        };
+        let habitat_name = manifest.habitat_name;
+
+        Ok(Chart {
+            name,
+            habitat_name,
+            chartfile,
+            manifest_template,
+            values,
+            ui,
+        })
+    }
+
+    pub fn generate(&mut self) -> Result<()> {
+        self.ui.status(
+            Status::Creating,
+            format!("chart directory `{}`", self.name),
+        )?;
+        Chart::create_dir(&self.name)?;
+
+        self.generate_chartfile()?;
+
+        let template_path = format!("{}/{}", self.name, "templates");
+        self.ui.status(
+            Status::Creating,
+            format!("templates directory `{}`", template_path),
+        )?;
+        Chart::create_dir(&template_path)?;
+        self.generate_manifest_template(&template_path)?;
+
+        self.generate_values()
+    }
+
+    pub fn generate_chartfile(&mut self) -> Result<()> {
+        let path = format!("{}/Chart.yaml", self.name);
+        self.ui.status(
+            Status::Creating,
+            format!("chart file `{}`", path),
+        )?;
+        let mut write = fs::File::create(path)?;
+        let out = self.chartfile.into_string()?;
+
+        write.write(out.as_bytes())?;
+
+        Ok(())
+    }
+
+    pub fn generate_manifest_template(&mut self, template_path: &str) -> Result<()> {
+        let manifest_path = format!("{}/{}.yaml", template_path, self.habitat_name);
+        self.ui.status(
+            Status::Creating,
+            format!("manifest template `{}`", manifest_path),
+        )?;
+        let mut write = fs::File::create(manifest_path)?;
+        let out = self.manifest_template.into_string()?;
+
+        write.write(out.as_bytes())?;
+
+        Ok(())
+    }
+
+    pub fn generate_values(&mut self) -> Result<()> {
+        let path = format!("{}/values.yaml", self.name);
+        self.ui.status(
+            Status::Creating,
+            format!("values file `{}`", path),
+        )?;
+        let mut write = fs::File::create(path)?;
+
+        self.values.generate(&mut write)?;
+
+        Ok(())
+    }
+
+    fn create_dir(path: &str) -> Result<()> {
```

Comment:
Maybe use `fs::create_dir_all` instead of adding this function?

Reply:
hmm.. true.